interface/controller/pokemon.go

```
@@ -79,6 +80,60 @@ func (pc *PokemonController) GetPokemon(res http.ResponseWriter, req *http.Reque
 
 }
 
+func (pc *PokemonController) GetPokemonsWorker(res http.ResponseWriter, req *http.Request) {
+	params := req.URL.Query()
+	t := params.Get("type")
+
+	if t == ""{
+		log.Println("ERROR: Empty value in query params type.")
+		http.Error(res, errors.New("empty value in query params type").Error(), http.StatusBadRequest)
+
+		return
+		
+	}
+
+	if t != "odd" &&  t!="even"{
+		log.Println("ERROR: Invalid type value in query params.", t)
+		http.Error(res, errors.New("invalid type value in query params").Error(), http.StatusBadRequest)
+
+		return
+	}
+
+	// Parsing the 'item' from string into int.
+	items, err := strconv.Atoi(params.Get("items"))
+	if err != nil {
+		log.Println("ERROR: Couldn't parse items Properly to Integer.")
+		http.Error(res, err.Error(), http.StatusBadRequest)
+
+		return
+	}
+
+	// Parsing the 'itemsWorker' from string into int.
+	itemsWorker, err := strconv.Atoi(params.Get("items_per_worker"))
+	if err != nil {
+		log.Println("ERROR: Couldn't parse itemsWorker Properly to Integer.")
+		http.Error(res, err.Error(), http.StatusBadRequest)
+
+		return
+	}
+
+	// Get pokemons using the WorkerPool
+	results := pc.PokemonInteractor.GetPokemonWorker(t, items, itemsWorker)
```

So, if `results` is empty, how can you know what's the reason for this?

The CSV doesn't exist? It didn't have any pokemon? Something else failed?

You should return an error alongside the results to better know what kind of response to return to the end user.